src/Rules/UseTypeHintForNewMethodsRule.php

```diff
@@ -0,0 +1,118 @@
+<?php
+/*
+ * Copyright (c) Since 2007 PrestaShop.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace PHPStanForPrestaShop\Rules;
+
+use PhpParser\Node;
+use PhpParser\Node\Stmt\ClassMethod;
+use PHPStan\Analyser\Scope;
+use PHPStan\Rules\Rule;
+use PHPStan\Rules\RuleErrorBuilder;
+use PHPStanForPrestaShop\PHPConfigurationLoader\ConfigurationLoaderInterface;
+
+/**
+ * @implements Rule<Node\Stmt\ClassMethod>
+ */
+class UseTypeHintForNewMethodsRule implements Rule
+{
+    /** @var array */
+    private $excludedClassMethodsList;
+
+    /**
+     * @param ConfigurationLoaderInterface $configurationFileLoader
+     */
+    public function __construct(ConfigurationLoaderInterface $configurationFileLoader)
+    {
+        $this->excludedClassMethodsList = $configurationFileLoader->load();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getNodeType(): string
+    {
+        return ClassMethod::class;
+    }
+
+    /**
+     * @param ClassMethod $node
+     * @param Scope $scope
+     *
+     * @return array
+     * @throws \PHPStan\ShouldNotHappenException
+     */
+    public function processNode(Node $node, Scope $scope): array
```

Comment:
The logic here follows this line:

1. check if function is part of the exclusion list, if yes, return nothing.
2. check if function has untyped parameters, if no, return nothing.
3. if class has parents, look at them and see if they have the same method ; if no, then it's a rule violation: there are untyped parameters.
4. if class has parents who also have this class method, see if they are part of the exclusion list, if yes, return nothing.
5. return violation else.

Reply:
Maybe instead of writing comments here, it should be inside the phpdoc? :trollface:

Reply:
I hesitated to do it, because it kindof duplicate the logic. Once in the code, once in the phpDoc. I was half pro, half against. Do you think I should do it ?

Reply:
Put in between code lines

Reply:
If you have to define it in a comment inside a PR, I think this is something you can add in the code. Otherwise, only in the PR description.
This comment will be lost if I clean on "Resolve conversation" :)

Reply:
Done in https://github.com/PrestaShop/phpstan-prestashop/pull/11/commits/4618f8a3ed3b35c5ff7199e61adce18bf0c79ed1